File: 剑指Offer/旋转数组的最小数字/minOfRotateArray.py
# -*- coding:utf-8 -*-
class Solution:
    def minNumberInRotateArray(self, rotateArray):
        # write code here
        left,right = 0,len(rotateArray)-1
        mid = -1
        while rotateArray[left]>=rotateArray[right]:
            if right - left == 1:
                return rotateArray[right]
            mid = left + (right-left)//2
            if rotateArray[mid] >= rotateArray[left]:
                left = mid
            if rotateArray[mid] <= rotateArray[right]:
                right = mid
        return rotateArray[mid]


# -*- coding:utf-8 -*-
class Solution:
    def minNumberInRotateArray(self, rotateArray):
        # write code here
        rotateArray.sort()
        return rotateArray[0]

File: 剑指Offer/旋转数组的最小数字/minOfRotateArray.java
import java.util.ArrayList;
public class Solution {
    public int minNumberInRotateArray(int [] array) {
        if(array.length == 0){
            return 0;
        }
        int left = 0;
        int right = array.length-1;
        int mid = -1;
        while(array[left]>=array[right]){
            if(right - left == 1){
                mid = right;
                break;
            }
            mid = left + (right - left )/2;
            if(array[mid]>=array[left]){
                left = mid;
            }
            if(array[mid]<=array[right]){
                right = mid;
            }
        }
        return array[mid];
    }
}

import java.util.ArrayList;
import java.util.Arrays;
public class Solution {
    public int minNumberInRotateArray(int [] array) {
        Arrays.sort(array);
        return array[0];
    }
}